onionbalance/hs_v3/service.py

```diff
@@ -263,6 +263,28 @@ def publish_descriptors(self):
         self._publish_descriptor(is_first_desc=True)
         self._publish_descriptor(is_first_desc=False)
 
+    
+    def _get_intros_for_distinct_desc(self, maxCount=10):
+        """
+        Get the intros that should be included in a descriptor for this service.
+        """
+        all_intros = self.get_all_intros_for_publish()
+
+        # Get number of instances that contributed to final intro point list
+        n_instances = len(all_intros.intro_points)
+
+        final_intros = all_intros.choose(maxCount)
+
+        if (len(final_intros) == 0):
+            logger.info("Got no usable intro points from our instances. Delaying descriptor push...")
+            raise NotEnoughIntros
+
+        # logger.info("We got %d intros from %d instances. We want %d intros ourselves (got: %d)",
```

Comment:
This log message got removed but it was useful. We should reinstall it somewhere.